File: LeetCode/2149/solution.java
// 2149. Rearrange Array Elements by Sign
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] ans = new int[nums.length];

        int pos=0, neg=1;
        for (int num: nums) {
            if (num > 0) {
                ans[pos] = num;
                pos += 2;
            }
            if(num < 0) {
                ans[neg] = num;
                neg += 2;
            }
        }
        
        return ans;
    }
}

File: LeetCode/2149/solution.cpp
// 2149. Rearrange Array Elements by Sign
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> ans(nums.size());

        int pos=0, neg=1;
        for(int& a: nums){
            if(a > 0){
                ans[pos] = a;
                pos += 2;
            }
            if(a < 0){
                ans[neg] = a;
                neg += 2;
            }
        }

        return ans;
    }
};